sig-security/joint-review.md

```diff
@@ -0,0 +1,375 @@
+- The Edged part of EdgeCore handles the running of Pods. Edged handles the running of Pods. The containers in these pods have low authority over the KubeEdge system and privileges thus flow from low to high. An attacker in control of some components in the containers should not be able to grow a foothold of more of the KubeEdge system.
+
+> Note: The MetaServer of component EdgeCore will start a Secure HTTPS Server, provide mutual certificate authentication and RBAC authentication to ensure communication security. This was hardened after the security audit.
+
+(This section is referenced from the security audit report under section `KubeEdge attack surface enumeration`).
+
+### Expected Attacker Capabilities
+
+All internal components of KubeEdge use TLS to communicate with each other, there are currently no known vulnerabilities and operators have the ability to exploit TLS for the communication between internal components. As for the components (ServiceBus of EdgeCore) that communicate with external applications, TLS is not enforced, but the service only exposed to the local host and the KubeEdge team has documented the risks of an insecure setup and made it harder for operators to deploy insecurely.
+
+Areas that are potentially susceptible to attacks would be credential handling, for example, token-based single sign-on from edge side to cloud side. However, all of these use some form of token based authentication and we assume the attacker cannot easily break existing cryptographic algorithms like AES or SHA256.
+
+### Attack Risks and Effects
+
+- Data saved in one database on the edge node, if an attacker attacks an edge database, it will cause denial of service in off-line scenarios.
+- External App can request Kubernetes resource to Meta-Server through list-watch method, then the request will be passed to CloudCore and finally to the api-server of the cluster. If an app was exploited by attacker, a DOS attack to list-watch server can be launched.
+
+### Security Degradation
+
+Exploiting vulnerabilities in management on the cloud can lead to maliciously deleting resources or bringing the whole cluster down. If an attacker intercepts tokens used for communicating with CloudCore, he can register a fake node and render the CloudCore unusable. If attackers gain access to the EdgeCore database, he could manipulate critical metadata such as adding or deleting existing pods or change the host mode or image pull address of pods.
+
+### Compensating Mechanisms
+
+- Please see the mitigations described at [here](https://github.com/kubeedge/community/blob/master/sig-security/sig-security-audit/KubeEdge-threat-model-and-security-protection-analysis.md#kubeedge-threat-model-and-mitigations), under the section `KubeEdge threat-model and mitigations`.

```

Comment:
It's recommended to include brief summary of the "threat-model and mitigations" here, e.g. number of risks and mitigations provided.

Reply:
Also, instead of `[here]`, I'd suggest to use the document name